aiplatform/snippets/embeddings_test.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package snippets
+
+import (
+	"testing"
+
+	"github.com/GoogleCloudPlatform/golang-samples/internal/testutil"
+)
+
+func TestGenerateEmbeddings(t *testing.T) {
+	tc := testutil.SystemTest(t)
+	apiEndpoint := "us-central1-aiplatform.googleapis.com:443"
+	model := "textembedding-gecko@003"
+	texts := []string{"banana muffins? ", "banana bread? banana muffins?"}
+	embeddings, err := embedTexts(apiEndpoint, tc.ProjectID, model, texts, "RETRIEVAL_DOCUMENT")
+	if err != nil {
+		t.Fatal(err)
+	}
+	if len(embeddings) != len(texts) || len(embeddings[0]) != 768 {
+		t.Errorf("len(embeddings), len(embeddings[0]) = %d, %d, want %d, 768", len(embeddings), len(embeddings[0]), len(texts))
+	}
+}
+
+func TestGenerateEmbeddingsPreview(t *testing.T) {
+	tc := testutil.SystemTest(t)
+	apiEndpoint := "us-central1-aiplatform.googleapis.com:443"
```

Comment:
ditto (:443)

Reply:
In Java and Go-lang, you will be given this error message:

rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing: dial tcp: address us-central1-aiplatform.googleapis.com: missing port in address"